core-spi-test/src/main/java/org/ehcache/internal/store/StoreReplaceKeyValueValueTest.java

```diff
@@ -151,7 +151,7 @@ public void wrongKeyTypeThrowsException()
 
     try {
       if (this.factory.getKeyType() == String.class) {
```

Comment:
Not your code, but this type check is wrong, it should be: `this.factory.getKeyType().isAssignableFrom(String.class)`